internal/wire/ack_frame.go

```diff
@@ -215,23 +190,29 @@ func (f *AckFrame) validateAckRanges() bool {
 	return true
 }
 
+// LargestAcked is the largest acked packet number
+func (f *AckFrame) LargestAcked() protocol.PacketNumber {
+	return f.AckRanges[0].Largest
+}
+
+// LowestAcked is the lowest acked packet number
+func (f *AckFrame) LowestAcked() protocol.PacketNumber {
+	return f.AckRanges[len(f.AckRanges)-1].Smallest
+}
+
 // AcksPacket determines if this ACK frame acks a certain packet number
 func (f *AckFrame) AcksPacket(p protocol.PacketNumber) bool {
-	if p < f.LowestAcked || p > f.LargestAcked { // this is just a performance optimization
+	if p < f.LowestAcked() || p > f.LargestAcked() { // this is just a performance optimization
 		return false
 	}
 
-	if f.HasMissingRanges() {
-		// TODO: this could be implemented as a binary search
-		for _, ackRange := range f.AckRanges {
-			if p >= ackRange.Smallest && p <= ackRange.Largest {
-				return true
-			}
+	// TODO: this could be implemented as a binary search
```

Comment:
There's a binary search implementation in the stdlib btw: https://golang.org/pkg/sort/#Search

Reply:
Will implement that in a separate PR.